src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -908,7 +908,7 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
                     pattern = rList[0];
                     if (rList.Count > 1)
                     {
-                        substitute = rList[1];
+                        substitute = PSObject.Base(rList[1]);
```

Comment:
Could you please add a comment why it is needed? what we break without this?

Reply:
Sure, This is actually a pretty fundamental operation in PowerShell. Anywhere you receive something of type `object`, you have to deal with the case where it might be wrapped in a `PSObject`. So before you can use it as `string` or `ScriptBlock`, you have to get the base object. The `PSObject.Base()` method encapsulates this operation. If it's passed a PSObject, it returns the core object otherwise it just returns the object.